File: program/code/InteractiveEntry.java
public class InteractiveEntry {
    public static final int QUIT = 0;
    public static final int OPTION_6 = 6;

    /*
     * NAME: entry
     * IMPORT(S): NONE
     * EXPORT(S): NONE
     * PURPOSE: Provide entry point for Interactive operations
     * CREATION: 03/10/2020
     * LAST MODIFICATION: 07/10/2020
     */

    public static void entry() {
        Set<String> excludedAssets = new Set<String>();
        int selection;
        do {
            System.out.println("Interactive Menu\n\n" + 
                "1. Asset Details\n" + 
                "2. Trade Details\n" +
                "3. Trade Paths\n" +
                "4. Asset Filter\n" +
                "5. Asset Overview\n" +
                "6. Trade Overview\n" +
                "0. Quit\n");
            String prompt = "Selection: ";
            selection = UserInterface.userInput(QUIT, OPTION_6, prompt);
            System.out.println();

            // Launches user-specifed operation
            if (selection != QUIT) {
                launch(selection, excludedAssets);
            }
        }
        while (selection != QUIT);
    }

    /*
     * NAME: launch
     * IMPORT(S): selection (int)
     * EXPORT(S): NONE
     * PURPOSE: Launch Interactive operations
     * CREATION: 03/10/2020
     * LAST MODIFICATION: 07/10/2020
    */

    private static void launch(int selection, Set<String> excludedAssets) {
        switch (selection) {
            // User specified "Display Asset"
            case 1:
                InteractiveAnalyser.getAssetTradePairs();
                break;
            // User specified "Display Trade Details"
            case 2:
                InteractiveAnalyser.getTradePriceChangeInfo();
                break;
            // User specified "Display Trade Paths"
            case 3:
                InteractiveAnalyser.getAssetGraph();
                break;
            // User specified "Asset Filter"
            case 4:
                InteractiveAndReportAnalyser.changeExcludedAssets(excludedAssets); 
                break;
            // User specified "Asset Overview"
            case 5:
                InteractiveAnalyser.getAllTradePairs(excludedAssets);
                break;
            // User specified "Trade Overview"
            case 6:
                InteractiveAnalyser.getRecentTrades();
                break;
        }
    }    
}

File: python/program/code/InteractiveEntry.py
"""
NAME: InteractiveEntry
AUTHOR: Tanaka Chitete
PURPOSE: Act as entry point for Taurus (Interactive Mode)
CREATION: 03/03/2021
LAST MODIFICATION: 04/03/2021
"""

import RecentTrades
import MarketInfo
import Set
import TradePairs
import TradePaths
import UserInterface

"""
NAME: menu
IMPORT(S): None
EXPORT(S): None
PURPOSE: Print menu and prepare to launch user-specified operation
CREATION: 03/03/2021
LAST MODIFICATION: 04/03/2021
"""

def menu():
    assetFilter = Set()
    userInput = None
    while True:
        print("Taurus (Interactive Mode)\n\n" + \
            "1. Get and Display Market Information of Trade Pair\n" + \
            "2. Get and Display Recent Trades (sorted by price, quantity and quote)\n" + \
            "3. Get and Display All Trade Paths Between Two Crypto-currencies\n" + \
            "4. Get and Display All Trade Pairs Involving Crypto-currency\n" + \
            "5. Get and Display Crypto-Filtered Trade Pairs\n" + \
            "6. Configure Crypto Filter\n" + \
            "0. Quit\n"
        )   
        prompt = "Selection: "
        userInput = UserInterface.getInt(0, 6, prompt)
        print() # Formatting purposes

        if userInput == 0:
            break
        else:
            launch(userInput, assetFilter)


"""
NAME: launch
IMPORT(S): userInput (int), assetFilter (Set)
EXPORT(S): None
PURPOSE: Launch user-specified operation
CREATION: 03/03/2021
LAST MODIFICATION: 04/03/2021
"""

def launch(userInput, assetFilter):
    if userInput == 1:
        MarketInfo.subMenu()
    elif userInput == 2:
        RecentTrades.subMenu()
    elif userInput == 3:
        TradePaths.subMenu()
    elif userInput == 4:
        TradePairs.subMenu()
    elif userInput == 5:
        FilteredTradePairs.subMenu(assetFilter)
    elif userInput == 6:
        AssetFilter.subMenu(assetFilter)